qiskit/transpiler/passes/routing/algorithms/token_swapper.py

```diff
@@ -91,157 +93,13 @@ def map(self, mapping: Mapping[int, int], trials: int = 4) -> list[Swap[int]]:
-                            predecessor
-                            for predecessor in digraph.predecessor_indices(unhappy_node)
-                            if predecessor != unhappy_node
-                        )
-                    except StopIteration:
-                        logger.error(
-                            "Unexpected StopIteration raised when getting predecessors"
-                            "in unhappy swap case."
-                        )
-                        return
-                    yield unhappy_node, predecessor
-                    swap(unhappy_node, predecessor)
-                    steps += 1
-        if todo_nodes:
-            raise RuntimeError("Too many iterations while approximating the Token Swaps.")
-
-    def _add_token_edges(
-        self, node: int, tokens: Mapping[int, int], digraph: rx.PyDiGraph, sub_digraph: rx.PyDiGraph
-    ) -> None:
-        """Add diedges to the graph wherever a token can be moved closer to its destination."""
-        if node not in tokens:
-            return
-
-        if tokens[node] == node:
-            digraph.extend_from_edge_list([(node, node)])
-            return
-
-        for neighbor in self.graph.neighbors(node):
-            if self.distance(neighbor, tokens[node]) < self.distance(node, tokens[node]):
-                digraph.extend_from_edge_list([(node, neighbor)])
-                sub_digraph.extend_from_edge_list([(node, neighbor)])
-
-    def _swap(
-        self,
-        node1: int,
-        node2: int,
-        tokens: MutableMapping[int, int],
-        digraph: rx.PyDiGraph,
-        sub_digraph: rx.PyDiGraph,
-        todo_nodes: MutableSet[int],
-    ) -> None:
-        """Swap two nodes, maintaining the data structures."""
-        assert self.graph.has_edge(
-            node1, node2
-        ), "The swap is being performed on a non-existent edge."
-        # Swap the tokens on the nodes, taking into account no-token nodes.
-        token1 = tokens.pop(node1, None)
-        token2 = tokens.pop(node2, None)
-        if token2 is not None:
-            tokens[node1] = token2
-        if token1 is not None:
-            tokens[node2] = token1
-        # Recompute the edges incident to node 1 and 2
-        for node in [node1, node2]:
-            digraph.remove_edges_from(
-                [(node, successor) for successor in digraph.successor_indices(node)]
-            )
-            sub_digraph.remove_edges_from(
-                [(node, successor) for successor in sub_digraph.successor_indices(node)]
-            )
-            self._add_token_edges(node, tokens, digraph, sub_digraph)
-            if node in tokens and tokens[node] != node:
-                todo_nodes.add(node)
-            elif node in todo_nodes:
-                todo_nodes.remove(node)
+        # Since integer seed is used in rustworkx, take random integer from np.random.randint
+        # and use that for the seed.
+        seed = np.random.randint(1, 10000, size=1)[0]
```

Comment:
I think we probably should do:

```suggestion
        seed = self.seed.integers(1, 10000)
```

so that we're respecting the input seed to the class.

Reply:
Sure thing. Done in https://github.com/Qiskit/qiskit-terra/pull/10001/commits/39617ec6f96fcf561e7eef86cf1d5b415ea23621.